frontend/app/.server/domain/repositories/client-eligibility.repository.ts

```diff
@@ -0,0 +1,101 @@
+import { inject, injectable } from 'inversify';
+import { None, Option, Some } from 'oxide.ts';
+
+import type { ServerConfig } from '~/.server/configs';
+import { TYPES } from '~/.server/constants';
+import type { ClientEligibilityEntity, ClientEligibilityRequestEntity } from '~/.server/domain/entities/client-eligibility.entity';
+import type { HttpClient } from '~/.server/http';
+import type { Logger } from '~/.server/logging';
+import { createLogger } from '~/.server/logging';
+import clientEligibilityJsonDataSource from '~/.server/resources/power-platform/client-eligibility.json';
+
+export interface ClientEligibilityRepository {
+  /**
+   * Finds client eligibility by client number.
+   * @param clientEligibilityRequestEntity The request object containing the client number.
+   * @returns A Promise that resolves to the client eligibility entity if found, or None otherwise.
+   */
+  findClientEligibilityByClientNumbers(clientEligibilityRequestEntity: ClientEligibilityRequestEntity): Promise<Option<Array<ClientEligibilityEntity>>>;
+}
+
+@injectable()
+export class DefaultClientEligibilityRepository implements ClientEligibilityRepository {
+  private readonly log: Logger;
+  private readonly serverConfig: Pick<ServerConfig, 'INTEROP_API_BASE_URI' | 'HTTP_PROXY_URL' | 'INTEROP_API_SUBSCRIPTION_KEY'>;
+  private readonly httpClient: HttpClient;
+
+  constructor(
+    @inject(TYPES.ServerConfig)
+    serverConfig: Pick<ServerConfig, 'INTEROP_API_BASE_URI' | 'HTTP_PROXY_URL' | 'INTEROP_API_SUBSCRIPTION_KEY'>,
+    @inject(TYPES.HttpClient)
+    httpClient: HttpClient,
+  ) {
+    this.log = createLogger('DefaultClientEligibilityRepository');
+    this.serverConfig = serverConfig;
+    this.httpClient = httpClient;
+  }
+
+  async findClientEligibilityByClientNumbers(clientEligibilityRequestEntity: ClientEligibilityRequestEntity): Promise<Option<Array<ClientEligibilityEntity>>> {
+    this.log.trace('Fetching client eligibility for client numbers [%j]', clientEligibilityRequestEntity);
+
+    const url = new URL(`${this.serverConfig.INTEROP_API_BASE_URI}/dts/applicant-eligibility`);
+    const response = await this.httpClient.instrumentedFetch('http.client.interop-api.applicant-eligibility_by-client-number.posts', url, {
+      proxyUrl: this.serverConfig.HTTP_PROXY_URL,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Ocp-Apim-Subscription-Key': this.serverConfig.INTEROP_API_SUBSCRIPTION_KEY,
+      },
+      body: JSON.stringify(clientEligibilityRequestEntity),
+    });
+
+    if (response.status === 200) {
```

Comment:
Return response if the response is `ok`.
```suggestion
    if (response.ok) {
```